components/tor/tor_file_watcher.h

```diff
@@ -0,0 +1,72 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_TOR_TOR_FILE_WATCHER_H_
+#define BRAVE_COMPONENTS_TOR_TOR_FILE_WATCHER_H_
+
+#include <memory>
+#include <vector>
+
+#include "base/callback.h"
+#include "base/files/file_path.h"
+#include "base/files/file_path_watcher.h"
+#include "base/memory/weak_ptr.h"
+
+namespace base {
+class SequencedTaskRunner;
+struct OnTaskRunnerDeleter;
+}  // namespace base
+
+namespace tor {
+
+FORWARD_DECLARE_TEST(TorFileWatcherTest, EatControlCookie);
+FORWARD_DECLARE_TEST(TorFileWatcherTest, EatControlPort);
+
+// This is used to fetch Tor cookie and port which are required to establish
+// control channel. It will delete itself when WatchCallback is called.
+class TorFileWatcher {
+ public:
+  using WatchCallback = base::OnceCallback<
+      void(bool success, std::vector<uint8_t> cookie, int port)>;
+
+  explicit TorFileWatcher(const base::FilePath& watch_dir_path);
+
+  virtual ~TorFileWatcher();
```

Comment:
should it be virtual?

Reply:
remove in aeed04b506e574973c5df12c9781dde63e187342